themes/src/main/resources/theme/base/login/select-authenticator.ftl

```diff
@@ -2,34 +2,41 @@
 <@layout.registrationLayout displayInfo=true; section>
     <#if section = "header" || section = "show-username">
         <script type="text/javascript">
-            // Fill up the two hidden and submit the form
-            function fillAndSubmit() {
-                document.getElementById('authexec-hidden-input').value = document.getElementById('authenticators-choice').value;
+            function fillAndSubmit(authExecId) {
+                document.getElementById('authexec-hidden-input').value = authExecId;
                 document.getElementById('kc-select-credential-form').submit();
             }
-            <#if auth.authenticationSelections?size gt 1>
-                // We bind the action to the select
-                window.addEventListener('load', function() {
-                    document.getElementById('authenticators-choice').addEventListener('change', fillAndSubmit);
-                });
-            </#if>
         </script>
+        <#if section = "header">
+            ${msg("loginChooseAuthenticator")}
+        </#if>
     <#elseif section = "form">
+
         <form id="kc-select-credential-form" class="${properties.kcFormClass!}" action="${url.loginAction}" method="post">
-            <div class="${properties.kcFormGroupClass!}">
-                <div class="${properties.kcLabelWrapperClass!}">
-                    <label for="authenticators-choice" class="${properties.kcLabelClass!}">${msg("loginCredential")}</label>
-                </div>
-                <div class="${properties.kcInputWrapperClass!}">
-                    <select id="authenticators-choice" class="form-control" size="1">
-                        <#list auth.authenticationSelections as authenticationSelection>
-                            <option value="${authenticationSelection.authExecId}" <#if authenticationSelection.authExecId == execution>selected</#if>>${msg('${authenticationSelection.authExecDisplayName}')}</option>
-                        </#list>
-                    </select>
-                    <input type="hidden" id="authexec-hidden-input" name="authenticationExecution" />
-                </div>
+            <div class="${properties.kcSelectAuthListClass!}">
+                <#list auth.authenticationSelections as authenticationSelection>
+                    <div class="${properties.kcSelectAuthListItemClass!}">
+                        <div class="${properties.kcSelectAuthListItemInfoClass!}" onclick="fillAndSubmit('${authenticationSelection.authExecId}')">
```

Comment:
@stianst also pointed this. We discuss it will be good to create follow-up JIRA for this to avoid javascript. It will be ideal to do this as a follow-up to not block this PR.

Reply:
Sounds good. Created https://issues.redhat.com/browse/KEYCLOAK-12823